lib/pwned_password_downloader.rb

```diff
@@ -0,0 +1,112 @@
+#!/usr/bin/env ruby
+require 'fileutils'
+require 'net/http/persistent'
+require 'retries'
+require 'ruby-progressbar'
+require 'ruby-progressbar/outputs/null'
+
+class PwnedPasswordDownloader
+  attr_reader :destination,
+              :num_threads,
+              :keep_threshold,
+              :output_progress
+
+  alias_method :output_progress?, :output_progress
+
+  RANGE_API_ROOT = 'https://api.pwnedpasswords.com/range/'
+  SHA1_LENGTH = 40
+  HASH_PREFIX_LENGTH = 5
+  OCCURRENCE_OFFSET = SHA1_LENGTH - HASH_PREFIX_LENGTH + ':'.length
+
+  def initialize(
+    destination: 'tmp/pwned',
+    num_threads: 64,
+    keep_threshold: 30,
+    output_progress: true
+  )
+    @destination = destination
+    @num_threads = num_threads
+    @keep_threshold = keep_threshold
+    @output_progress = output_progress
+  end
+
+  def run!(start: '00000', finish: 'FFFFF')
```

Comment:
Yeah, I'd thought about that. I considered adding a "demo mode" flag to the `pwned-passwords.sh` script to support downloading a smaller fragment of the full download. 

The tricky thing is figuring how best to pass those parameters into the script, since right now it's executed in a way that doesn't expect any options to be passed.

https://github.com/18F/identity-idp/blob/1c05cb5f415a98a798b5a59995a42417d0c8305f/scripts/pwned-passwords.sh#L25

Could use environment variables, or swap that line of code with something more like...

```bash
bundle exec ruby -e "require_relative 'lib/pwned_password_downloader'; PwnedPasswordDownloader.new.run!(start: '$start', finish: '$finish')"
```

Reply:
I think I'll leave it out for this go-around.